db/view/build_progress_virtual_reader.hh

```diff
@@ -120,6 +120,21 @@ class build_progress_virtual_reader {
             return clustering_key_prefix::from_exploded(r);
         }
 
+        position_in_partition adjust_ckey(position_in_partition&& underlying_pip) {
+            auto& underlying_ck_opt = underlying_pip.get_clustering_key_prefix();
+            if (!underlying_ck_opt || !underlying_ck_opt->is_full(underlying_schema())) {
+                return std::move(underlying_pip);
+            }
+            auto& underlying_ck = underlying_ck_opt.value();
+            // Drop the cpu_id from the clustering key prefix (which we know to be full)
+            auto end = underlying_ck.begin(underlying_schema());
+            std::advance(end, underlying_schema().clustering_key_size() - 1);
+            auto r = boost::make_iterator_range(underlying_ck.begin(underlying_schema()), std::move(end));
```

Comment:
Can't we reuse the existing overload for this part?

Reply:
no - clustering_ck_prefix vs clustering_key.

Reply:
The two are the same:
https://github.com/scylladb/scylla/blob/dc886d96d1774508e18e0139113cb0cc9b181b2e/replica/database_fwd.hh#L54

Reply:
Ah! will merge.